core/trino-main/src/main/java/io/trino/execution/scheduler/PipelinedQueryScheduler.java

```diff
@@ -844,13 +845,13 @@ public static DistributedStagesScheduler create(
         {
             DistributedStagesSchedulerStateMachine stateMachine = new DistributedStagesSchedulerStateMachine(queryStateMachine.getQueryId(), executor);
 
-            Map<PartitioningHandle, NodePartitionMap> partitioningCacheMap = new HashMap<>();
-            BiFunction<PartitioningHandle, Optional<Integer>, NodePartitionMap> partitioningCache = (partitioningHandle, partitionCount) ->
-                    partitioningCacheMap.computeIfAbsent(partitioningHandle, handle -> nodePartitioningManager.getNodePartitioningMap(
+            Map<PartitioningKey, NodePartitionMap> partitioningCacheMap = new HashMap<>();
```

Comment:
Can we make this change a preparatory commit?  This will make the actual change much simpler and cleaner as a separate commit.